server/database_fns.py

```diff
@@ -26,117 +28,51 @@ def real_dicts_to_python_dicts(real_dict_list):
-                    submit_time ASC;
-                """
-
-    conn = None
-    try:
-
-        conn = psycopg2.connect(**DB_KWARGS)  # Connect to DB
-        cur = conn.cursor(cursor_factory=psycopg2.extras.RealDictCursor)  # Get cursor
-        cur.execute(command)  # Send the command
-
-        # Returns just the top one. If there is no queued, this
-        # value should be None.
-        rows = cur.fetchall()
-
-        cur.close()  # close communication with the PostgreSQL database server
-        conn.commit()  # commit the changes
-
-        return real_dicts_to_python_dicts(rows)
-
-    except (Exception, psycopg2.DatabaseError) as error:
-        print(error)
-    finally:
-        if conn is not None:
-            conn.close()
-
-
-def get_all_running():
+def get_all_jobs_by_status(status):
     """
 
-    Get all running jobs from the DB, sorted by
-    earliest submit_time. Returns them as a list of dicts.
-
-    """
-
-    command = """
-                SELECT
-                    *
-                FROM
-                    jobs
-                WHERE
-                    status = 'RUNNING'
-                ORDER BY
-                    start_time ASC;
-                """
-
-    conn = None
-    try:
-
-        conn = psycopg2.connect(**DB_KWARGS)  # Connect to DB
-        cur = conn.cursor(cursor_factory=psycopg2.extras.RealDictCursor)  # Get cursor
-        cur.execute(command)  # Send the command
-
-        # Returns just the top one. If there is no queued, this
-        # value should be None.
-        rows = cur.fetchall()
-
-        cur.close()  # close communication with the PostgreSQL database server
-        conn.commit()  # commit the changes
-
-        return real_dicts_to_python_dicts(rows)
-
-    except (Exception, psycopg2.DatabaseError) as error:
-        print(error)
-    finally:
-        if conn is not None:
-            conn.close()
-
-
-def get_all_completed():
-    """
-
-    Get all completed jobs from the DB, sorted by
-    earliest submit_time. Returns them as a list of dicts.
+    Get all jobs of one type from the DB, sorted by earliest submit_time.
+    Returns them as a list of dicts.
 
     """
 
+    assert (
+        status in VALID_STATUSES
+    ), f"Must provide a valid status! Provided = {status}"
+
+    # For COMPLETED, only want the most recent
+    status_sort_order = {
+        "QUEUED": "ASC",
+        "RUNNING": "ASC",
+        "COMPLETED": "DESC",
+        "FAILED": "DESC",
+    }
+
+    n_lim = 20  # The number of jobs to get
     command = """
                 SELECT
                     *
                 FROM
                     jobs
                 WHERE
-                    status = 'COMPLETED'
+                    status = '{}'
                 ORDER BY
-                    end_time ASC;
-                """
+                    submit_time {}
+                LIMIT {};
+                """.format(
+        status, status_sort_order[status], n_lim
+    )
```

Comment:
Can these have the {status} and {submit_time} and {n_limit} and have the formats use their corresponding bits.